source/ed/connectors/speed_parser.cpp

```diff
@@ -0,0 +1,172 @@
+    parser.max_speeds_by_type["primary_link"] = urban_speed;
+    parser.max_speeds_by_type["secondary_link"] = urban_speed;
+    parser.max_speeds_by_type["tertiary_link"] = urban_speed;
+    parser.max_speeds_by_type["unclassified"] = urban_speed;
+    parser.max_speeds_by_type["road"] = urban_speed;
+    parser.max_speeds_by_type["residential"] = urban_speed;
+    parser.max_speeds_by_type["living_street"] = living_street_speed;
+    parser.max_speeds_by_type["service"] = urban_speed;
+    parser.max_speeds_by_type["track"] = walking_speed;
+    parser.max_speeds_by_type["path"] = walking_speed;
+    parser.max_speeds_by_type["unclassified"] = urban_speed;
+    // useless but prevent warnings
+    parser.max_speeds_by_type["footway"] = walking_speed;
+    parser.max_speeds_by_type["steps"] = walking_speed;
+    parser.max_speeds_by_type["pedestrian"] = walking_speed;
+    parser.max_speeds_by_type["cycleway"] = walking_speed;
+    parser.max_speeds_by_type["bus_stop"] = walking_speed;
+    parser.max_speeds_by_type["platform"] = walking_speed;
+    parser.max_speeds_by_type["construction"] = walking_speed;
+    parser.max_speeds_by_type["bridleway"] = walking_speed;
+
+    parser.speed_factor_by_type["motorway"] = 0.9;
+    parser.speed_factor_by_type["motorway_link"] = 0.5;
+    parser.speed_factor_by_type["trunk"] = 0.8;
+    parser.speed_factor_by_type["trunk_link"] = 0.5;
+    parser.speed_factor_by_type["primary"] = 0.8;
+    parser.speed_factor_by_type["secondary"] = 0.6;
+    parser.speed_factor_by_type["tertiary"] = 0.6;
+    parser.speed_factor_by_type["primary_link"] = 0.5;
+    parser.speed_factor_by_type["secondary_link"] = 0.5;
+    parser.speed_factor_by_type["tertiary_link"] = 0.5;
+    parser.speed_factor_by_type["unclassified"] = 0.5;
+    parser.speed_factor_by_type["road"] = 0.5;
+    parser.speed_factor_by_type["residential"] = 0.5;
+    parser.speed_factor_by_type["living_street"] = 0.5;
+    parser.speed_factor_by_type["service"] = 0.2;
+    parser.speed_factor_by_type["track"] = 0.2;
+    parser.speed_factor_by_type["path"] = 0.2;
+    parser.speed_factor_by_type["unclassified"] = 0.5;
+    // useless but prevent warnings
+    parser.speed_factor_by_type["footway"] = 1;
+    parser.speed_factor_by_type["steps"] = 1;
+    parser.speed_factor_by_type["pedestrian"] = 1;
+    parser.speed_factor_by_type["cycleway"] = 1;
+    parser.speed_factor_by_type["bus_stop"] = 1;
+    parser.speed_factor_by_type["platform"] = 1;
+    parser.speed_factor_by_type["construction"] = 1;
+    parser.speed_factor_by_type["bridleway"] = 1;
+
+    return parser;
+}
+
+boost::optional<float> SpeedsParser::get_speed(const std::string& highway,
+                                               const boost::optional<std::string>& max_speed) const {
+    boost::optional<float> speed;
+    boost::optional<float> factor;
+
+    if (max_speeds_by_type.empty() && speed_factor_by_type.empty()) {
+        return boost::none;
```

Comment:
Shouldn't we return max_speed if a value is passed to get_speed even if they are both empty?
Maybe this check needs to go below `if (max_speed)`?

Reply:
This is a way to indicate that the speed parser hasn't been initialized so speed aren't imported. I can add a boolean to make things clear

Reply:
Ok for me then.
If we can configure the speed parser in the future there is a stupid case where you would pass empty maps for both of them and could still want to import maxspeed datas with no default value and only with the default speed_factor. But it's stupid :).